homeassistant/components/dsmr_reader/sensor.py

```diff
@@ -34,15 +30,10 @@ def __init__(self, topic):
         self._entity_id = slugify(topic.replace("/", "_"))
         self._topic = topic
 
-        self._name = self._definition["name"]
-        self._unit_of_measurement = (
-            self._definition["unit"] if "unit" in self._definition else ""
-        )
-        self._icon = self._definition["icon"] if "icon" in self._definition else None
-        self._transform = (
-            self._definition["transform"] if "transform" in self._definition else None
-        )
-
+        self._name = self._definition.get("name")
```

Comment:
Shouldn't we consider name required in the definition?

Reply:
Yes, we should, I now take the last part of the topic and use that as a default, when no name is available.